packages/test/test-end-to-end-tests/src/test/gc/gcUnreferencedTimestamp.spec.ts

```diff
@@ -132,165 +132,461 @@ describeFullCompat("GC unreferenced timestamp", (getTestObjectProvider) => {
-        const dsATimestamp2 = timestamps2.get(dataStoreA.id);
-        assert(dsATimestamp2 !== undefined, `data store should still have unreferenced timestamp`);
-        assert(dsATimestamp2 > dsATimestamp1, `new timestamp should be greater that the previous one`);
-    // This test has increased timeout because it waits for multiple summaries to be uploaded to server. It then also
-    // waits for those summaries to be ack'd. This may take a while.
-    }).timeout(20000);
+    describe("unreferenced timestamp in in summary", () => {
+        it("adds / removes unreferenced timestamp from data stores correctly", async () => {
+            const summarizerClient = await getNewSummarizer();
+
+            // Create a new data store and mark it as referenced by storing its handle in a referenced DDS.
+            const dataStoreB = await dataObjectFactory.createInstance(dataStoreA.containerRuntime);
+            dataStoreA._root.set("dataStoreB", dataStoreB.handle);
+
+            // Validate that the new data store does not have unreferenced timestamp.
+            const timestamps1 = await getUnreferencedTimestamps(summarizerClient);
+            const dsBTimestamp1 = timestamps1.get(dataStoreB.id);
+            assert(dsBTimestamp1 === undefined, `new data store should not have unreferenced timestamp`);
+
+            // Mark the data store as unreferenced by deleting its handle from the DDS and validate that it now has an
+            // unreferenced timestamp.
+            dataStoreA._root.delete("dataStoreB");
+            const timestamps2 = await getUnreferencedTimestamps(summarizerClient);
+            const dsBTimestamp2 = timestamps2.get(dataStoreB.id);
+            assert(dsBTimestamp2 !== undefined, `data store should have unreferenced timestamp`);
+
+            // Perform some operations and generate another summary. Validate that the data store still has the same
+            // unreferenced timestamp.
+            dataStoreA._root.set("key", "value");
+            const timestamps3 = await getUnreferencedTimestamps(summarizerClient);
+            const dsBTimestamp3 = timestamps3.get(dataStoreB.id);
+            assert(dsBTimestamp3 !== undefined, `data store should still have unreferenced timestamp`);
+            assert.strictEqual(dsBTimestamp2, dsBTimestamp3, "unreferenced timestamp should not have changed");
+
+            // Mark the data store as referenced again and validate that the unreferenced timestamp is removed.
+            dataStoreA._root.set("dataStoreB", dataStoreB.handle);
+            // Validate that the data store does not have unreferenced timestamp after being referenced.
+            const timestamps4 = await getUnreferencedTimestamps(summarizerClient);
+            const dsBTimestamp4 = timestamps4.get(dataStoreB.id);
+            assert(dsBTimestamp4 === undefined, `data store should not have unreferenced timestamp anymore`);
+        }).timeout(20000);
```

Comment:
Please, no 20 second running tests! Please mock timeouts and figure out how to run it in 2 seconds.

Reply:
Same below

Reply:
This is not due to delays added in tests. The tests take long to run because it uploads summary to odsp and then waits for the ack. Each test does this multiple times so it takes pretty long. Is there a good way to reduce this time?